src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -164,6 +165,45 @@ public function getDecoratedService(): ?array
         return $this->decoratedService;
     }
 
+    /**
+     * Sets the tag that this service is decorating.
+     *
+     * When decorating a tag, this definition acts as a template to create decorator services
+     * for each service that has the specified tag.
+     *
+     * @param string|null $tag             The tag name to decorate, use null to remove tag decoration
+     * @param int         $priority        The priority of decoration
+     * @param int         $invalidBehavior The behavior when a tagged service is invalid
+     *
+     * @return $this
+     */
+    public function setDecoratedTag(?string $tag, int $priority = 0, int $invalidBehavior = ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE): static
```

Comment:
do we really need this?
I mean: a definition for tag decoration is a resource class, so we could maybe use addResourceTag instead?